src/remediations/fifi.js

```diff
@@ -191,18 +243,25 @@ exports.getConnectionStatus = async function (remediation, account) {
 
     _.forEach(systems, system => system.satelliteId = exports.getSatelliteId(system.facts));
 
+    await fetchRHCClientId(systems);
+
     const satellites = _(systems).groupBy('satelliteId').mapValues(systems => ({
         id: systems[0].satelliteId,
         // unique by ansible host i.e. if there are two systems with the same ansible identifier then
         // only pick on one of them as we wouldn't be able to tell them apart based on responses from Satellite
         systems: _(systems).sortBy('id').uniqBy(generator.systemToHost).value()
     })).values().value();
 
+    if (!_.isEmpty(satellites)) {
+        defineRHCEnabledExecutor(satellites);
+    }
+
     const sourceInfo = await sources.getSourceInfo(_(satellites).map('id').filter().value());
 
     _.forEach(satellites, satellite => {
         satellite.source = _.get(sourceInfo, satellite.id, null);
         satellite.receptor = getReceptor(satellite.source);
+        satellite.rhcStatus = fetchRHCStatus(satellite.systems);
```

Comment:
This looks very brittle. Say you have 5 sat-enabled hosts. Should one of them define rhc_client for some reason (e.g. test data, leftover from when the host was registered directly before moving under satellite, etc.) that one host is going to make the entire Satellite as rhc_connected. As a result, remediations will from there on consider the entire executor (Satellite) as RHC and use RHC for running stuff on any of the 5 hosts.

Instead, it should be clearly defined who (Satellite vs RHC) wins and if a host is initially put into a "Satellite bucket" remediations should not under no circumstances convert the whole bucket into RHC.